memoization.js

```diff
@@ -1,38 +1,26 @@
 'use strict';
 
-const smartSelector = (() => {
-  const getElementById = ((id) => {
-    let cache = {};
+const smartSelect = (() => {
+  const cache = {};
 
-    return (id) => {
-      if (id in cache) {
-        return cache[id];
-      } else {
-        let select = document.getElementById(id);
+  const querySelector = (query) => {
+    const format = query.charAt(0) === '#' ? query : `#${query}`;
+    const selection = document.querySelector(format);
```

Comment:
So this would always call `querySelector` this doesn't help with memoization since you're still making the call even if the element exists in your cache. You'd want to check the cache first then call this method if it doesn't exist.